tests/container-copy

```diff
@@ -0,0 +1,70 @@
+#!/bin/sh
+set -eu
+
+# Install and setup LXD.
+install_lxd
+lxd init --auto
+
+# Create a loop device that we can mount as a source for a dir storage pool.
+test_dir="$(mktemp -d XXX.test)"
+loopimg="$(mktemp "${test_dir}/XXX.img")"
+truncate -s 5G "${loopimg}"
+loopdev="$(losetup --show --find "${loopimg}")"
+
+# Begin Test.
+set -x
+
+# Launch a VM which will be the target of the copy.
+lxc launch "${TEST_IMG:-ubuntu-minimal-daily:22.04}" target --vm
+waitInstanceBooted target
+
+# Add the target VM as a remote LXD.
+target_addr="$(lxc ls -f csv -c 4 target | grep -v lxdbr | cut -d' ' -f1 | sed 's/"//g')"
+lxc exec target -- snap install lxd --channel "${LXD_SNAP_CHANNEL}"
+lxc exec target -- lxd init --auto
+lxc exec target -- lxc config set core.https_address=:8443
+
+# Use `lxc remote add` to generate client certs, in case they don't exist.
+! lxc remote add generate-certs not-a-url --accept-certificate 2>/dev/null || true
```

Comment:
```suggestion
# Try to add a bogus remote to force lxc into generating a client crt/key pair if none was found
if ! [ -e ~/snap/lxd/common/config/client.crt ]; then
  lxc remote add generate-certs not-a-url --accept-certificate 2>/dev/null || true
fi
```
Would that be good enough?

Reply:
> I'm now questioning my own decision of doing this as being able to cleanup properly feels important too.

Yeah cleanup doesnt normally take long either.

Reply:
>   lxc remote add generate-certs not-a-url --accept-certificate 2>/dev/null || true
I'm thinking this is going to fail if/when we fix https://github.com/canonical/lxd/issues/12414

I'm not quite following why we need this "fudge"?

Reply:
> Turns out it's quite fast (4-5 minutes) so my concern was ill-founded.

I'm happy with a VM for now too.

Reply:
> > lxc remote add generate-certs not-a-url --accept-certificate 2>/dev/null || true
> > I'm thinking this is going to fail if/when we fix [canonical/lxd#12414](https://github.com/canonical/lxd/issues/12414)
> 
> I'm not quite following why we need this "fudge"?

It's because the client cert needs to be added to the `target`'s trust for it to become a usable remote.
```
lxc file push ~/snap/lxd/common/config/client.crt target/root/
lxc exec target -- lxc config trust add test /root/client.crt
```

Having said that, it feels like a good case for a [trust token](https://documentation.ubuntu.com/lxd/en/latest/howto/server_expose/#authenticate-with-the-lxd-server).

Reply:
> > lxc remote add generate-certs not-a-url --accept-certificate 2>/dev/null || true
> > I'm thinking this is going to fail if/when we fix [canonical/lxd#12414](https://github.com/canonical/lxd/issues/12414)
> 
> I'm not quite following why we need this "fudge"?

The client uses `~/snap/lxd/common/config` or `~/.config/lxc` to store client certs. To be able to use a remote there has to be a keypair present here. It doesn't exist by default when you `snap install lxd`. When we use `lxc remote add` we automatically create that directory and populate it before actually performing the action of adding the remote. 

In this case we are using cert auth so the client cert needs to have been trusted by the other system before we call `lxc remote add`, so we have to use it once to generate the certs, then move the certs over and trust them on the other side, and use it again with the actual url to add the remote.

Reply:
@masnax a trust token would be simpler and avoid moving files around.